lib/msf/util/exe.rb

```diff
@@ -991,6 +991,51 @@ def self.to_vba(framework,code,opts = {})
     read_replace_script_template("to_mem.vba.template", hash_sub)
   end
 
+  def self.to_win32_vbsmem(framework, code, opts={})
+		set_template_default(opts, "vbsmem.vbs")
+               payload = Msf::Simple::Buffer::transform(code, fmt='js_le')
+               s = "sShellCode = Unescape(\"#{payload}\")\n"
+               vbs_template = ''
+               File.open(opts[:template], 'rb') { |fd|
+                       vbs_template = fd.read(fd.stat.size)
+               }
+               s << vbs_template
+               if (1) 
```

Comment:
You should be able to use the random identifier class to generate random identifiers and ensure there are no clashes etc. This file probably has the examples you need...